jme3-effects/src/main/java/com/jme3/water/WaterFilter.java

```diff
@@ -292,8 +303,20 @@ protected void cleanUpFilter(Renderer r) {
     @Override
     public void write(JmeExporter ex) throws IOException {
         super.write(ex);
+
         OutputCapsule oc = ex.getCapsule(this);
 
+        final Spatial reflectionScene = getReflectionScene();
+        final boolean needSaveReflectionScene = isNeedSaveReflectionScene();
+
+        oc.write(causticsTexture, "causticsTexture", null);
+        oc.write(heightTexture, "heightTexture", null);
+        oc.write(normalTexture, "normalTexture", null);
+        oc.write(foamTexture, "foamTexture", null);
```

Comment:
Don't save the textures data. save the texture path and load them in the read method
Also you'll have to do something about this https://github.com/jMonkeyEngine/jmonkeyengine/blob/master/jme3-effects/src/main/java/com/jme3/water/WaterFilter.java#L230
only load the default textures if the texture2D variables are null.